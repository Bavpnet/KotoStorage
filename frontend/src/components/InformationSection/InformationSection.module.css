.infoPar {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
}

.alignRight {
  align-items: flex-end;
  text-align: right;
}

.pars {
  margin-top: 73px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 85%;
  line-height: 31px;
}

.low {
  margin-top: 28px;
}

.high {
  margin-top: 84px;
}

.infoPar span {
  background-color: #5e564b;
  height: 3px;
  width: 85%;
}

.infoPar .par3 {
  margin-top: 28px;
  width: 85%;
}
