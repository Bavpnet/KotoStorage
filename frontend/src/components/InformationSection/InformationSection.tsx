import styles from './InformationSection.module.css'

type InformationSectionProps = {
  title: string
  alignment: 'left' | 'right'
  p1: string
  p2: string
  breakLine?: boolean
  p3?: string
}

export const InformationSection: React.FC<InformationSectionProps> = ({
  title,
  alignment,
  p1,
  p2,
  p3,
  breakLine
}) => {
  return (
    <section
      className={`${styles.infoPar} ${
        alignment === 'right' ? styles.alignRight : ''
      }`}
    >
      <h3>{title}</h3>
      <div className={styles.pars}>
        <p>{p1}</p>
        <br />
        <p>{p2}</p>
      </div>
      {breakLine && <span className={p3 ? styles.low : styles.high} />}
      {p3 && <p className={styles.par3}>{p3}</p>}
    </section>
  )
}
