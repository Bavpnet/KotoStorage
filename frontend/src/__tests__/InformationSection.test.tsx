import renderer from 'react-test-renderer'
import { InformationSection } from '../components/InformationSection'

describe('InformationSection', () => {
  it('renders correctly', () => {
    const title = 'Information'
    const p1 = 'This is the first paragraph.'
    const p2 = 'This is the second paragraph.'
    const p3 = 'This is the third paragraph.'
    const tree = renderer
      .create(
        <InformationSection
          title={title}
          p1={p1}
          p2={p2}
          p3={p3}
          alignment={'left'}
          breakLine={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
