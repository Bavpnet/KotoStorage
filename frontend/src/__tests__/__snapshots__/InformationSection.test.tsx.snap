// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationSection renders correctly 1`] = `
<section
  className="infoPar "
>
  <h3>
    Information
  </h3>
  <div
    className="pars"
  >
    <p>
      This is the first paragraph.
    </p>
    <br />
    <p>
      This is the second paragraph.
    </p>
  </div>
  <span
    className="low"
  />
  <p
    className="par3"
  >
    This is the third paragraph.
  </p>
</section>
`;
